### PR TITLE
list transactions in dashboard

### PR DESCRIPTION
**Pull request by octobocto**
help @torkelrogstad

How can I categorize the transactions correctly? And what data should we show?

- rpc: clean up peg out and send rpc calls
- providers+rpc: add transaction provider
- tabs/dashboard: do quick n dirty transaction list

### CODE REVIEW

File: lib/providers/transactions_provider.dart [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  // used for polling
+  late Timer _timer;
+
+  TransactionsProvider() {
+    fetch();
+    _startPolling();
+  }
+
+  // call this function from anywhere to refresh the balance
```
Comment (torkelrogstad) [inline lib/providers/transactions_provider.dart:24]: incorrect comment

File: lib/rpc/rpc.dart [hunk 2]
```diff
@@ -161,9 +163,108 @@ class RPCLive implements RPC {
...
   static const errWithdrawalNotFound = -101;
 }
+
+class Transaction {
+  final String account;
+  final String address;
+  final String category;
+  final double amount;
+  final String label;
+  final int vout;
```
Comment (torkelrogstad) [inline lib/rpc/rpc.dart:189]: do you write these out by hand, or do you have a tool for this?

**Comment by torkelrogstad** [comment]
Not too sure if we can categorize all that much. A few ideas: 

1. Look at the labels for addresses to determine if these are peg-in addresses. 
image

2. Looks like peg-outs get a special "WT" comment: https://github.com/LayerTwo-Labs/testchain/blob/ba78df157fcb9f85d898f65db43c2842ab9473ff/src/wallet/wallet.cpp#L4380-L4385
image

Note that both peg-ins and peg-outs have multiple outputs in the same TX. For example the withdrawal TX has two outputs, I think the second smaller output is related to BIP300/BIP301 logic. This should not be displayed in the TX list?

File: lib/rpc/rpc.dart (hunk 2 above)
Comment (octobocto) [reply lib/rpc/rpc.dart:189]: Have a tool, it’s called chatgpt

Comment (octobocto) [reply lib/rpc/rpc.dart:189]: The neatest tool of them all. I use no other tools for flutter

**Comment by octobocto** [comment]
we either categorize it completely, or not at all imo. starting with no categories

**Pull request merged**